channels/channel.mtg/viafree/chn_viafree.py

```diff
@@ -537,6 +536,8 @@ def create_video_item(self, result_set):
             item.set_date(date_info[0], date_info[1], date_info[2], time_info[0], time_info[1], 0)
 
         item.media_type = mediatype.EPISODE
+        if is_episodic:
+            item.set_season_info(season, episode)        
```

Comment:
`Local variable 'episode' might be referenced before assignment` is the error here. You will need to define `season` and `episode` as `None` or 0 inbetween line 488 and 489. 

Also: trailing spaces in this line.

Reply:
Changes line 539 "if is_episodic:" to:
if episode> 0 and season> 0:

Changes line 467 "is_episodic = False" to:
season = 0
episode = 0